tests-ui/tests/composables/graph/useCanvasInteractions.test.ts

```diff
@@ -1,126 +1,183 @@
+
+      const { handlePointer } = useCanvasInteractions()
 
-      const wheelEvent = new WheelEvent('wheel', {
-        deltaY: -100,
-        cancelable: true
-      })
+      // Create mock pointer event
+      const mockEvent = {
+        buttons: 1,
+        preventDefault: vi.fn(),
+        stopPropagation: vi.fn()
+      } as unknown as PointerEvent
 
-      canvasInteractions.handleWheel(wheelEvent)
+      // Test - should not throw
+      expect(() => handlePointer(mockEvent)).not.toThrow()
 
-      expect(mockCanvas.dispatchEvent).toHaveBeenCalled()
+      // Verify
+      expect(mockEvent.preventDefault).not.toHaveBeenCalled()
     })
+  })
 
-    it('should handle missing canvas gracefully', () => {
-      ;(app.canvas as any).canvas = null
-      mockSettingStore.get.mockReturnValue('standard')
+  describe('handleWheel', () => {
+    it('should forward ctrl+wheel events to canvas in standard nav mode', () => {
+      // Setup
+      mockGet.mockReturnValue('standard')
+
+      const { handleWheel } = useCanvasInteractions()
 
-      const wheelEvent = new WheelEvent('wheel', {
+      // Create mock wheel event with ctrl key
+      const mockEvent = {
         ctrlKey: true,
-        deltaY: -100
-      })
+        metaKey: false,
+        preventDefault: vi.fn()
+      } as unknown as WheelEvent
+
+      // Test
+      handleWheel(mockEvent)
 
-      expect(() => {
-        canvasInteractions.handleWheel(wheelEvent)
-      }).not.toThrow()
+      // Verify
+      expect(mockEvent.preventDefault).toHaveBeenCalled()
     })
-  })
 
-  describe('forwardEventToCanvas', () => {
-    it('should dispatch event to canvas element', () => {
-      const wheelEvent = new WheelEvent('wheel', {
-        deltaY: -100,
-        ctrlKey: true
-      })
+    it('should forward all wheel events to canvas in legacy nav mode', () => {
+      // Setup
+      mockGet.mockReturnValue('legacy')
+
+      const { handleWheel } = useCanvasInteractions()
+
+      // Create mock wheel event without modifiers
+      const mockEvent = {
+        ctrlKey: false,
+        metaKey: false,
+        preventDefault: vi.fn()
+      } as unknown as WheelEvent
 
-      canvasInteractions.forwardEventToCanvas(wheelEvent)
+      // Test
+      handleWheel(mockEvent)
 
-      expect(mockCanvas.dispatchEvent).toHaveBeenCalledWith(
-        expect.any(WheelEvent)
-      )
+      // Verify
+      expect(mockEvent.preventDefault).toHaveBeenCalled()
     })
 
-    it('should handle missing canvas gracefully', () => {
-      ;(app.canvas as any).canvas = null
+    it('should not prevent default for regular wheel events in standard nav mode', () => {
+      // Setup
+      mockGet.mockReturnValue('standard')
+
+      const { handleWheel } = useCanvasInteractions()
+
+      // Create mock wheel event without modifiers
+      const mockEvent = {
+        ctrlKey: false,
+        metaKey: false,
+        preventDefault: vi.fn()
+      } as unknown as WheelEvent
```

Comment:
How would you feel about `satisfies Partial<WheelEvent>` as a small increase in type safety?